cpp/src/io/csv/csv_gpu.h

```diff
@@ -23,6 +23,99 @@ namespace cudf {
 namespace io {
 namespace csv {
 namespace gpu {
+
```

Comment:
This should probably be a .cuh file since it includes `__device__` code.

Reply:
> This should probably be a .cuh file since it includes `__device__` code.

It's exposing host inline functions, but they also happen to be also be used on the device side (__host__ __device__). It's not actually device-specific (everything in this file is meant to be usable on the cpp side) ~ not sure if there is a way other than `__host__ __device__` declaration to make normal code run on either cpp or cu ?

Reply:
Actually, I might just move these, along with the other __device__ functions that are not CSV-specific to `parsing_utils.cuh`, since they will likely be needed for the equivalent optimization in the JSON parser (which would allow us to get rid of the `{find|count}_all_from_set` and related utils

Reply:
Meh, looks like it's going to be a mess due to some things being under the `experimental` namespaces and others not (`cudf::io::gpu`, `cudf::io::csv::gpu`, `cudf::experimental::io::gpu` etc) -> will wait until the `experimental` namespace is removed when tackling the JSON reader.